packages/mobile/src/transactions/feed/queryHelper.ts

```diff
@@ -0,0 +1,154 @@
+import { useAsync } from 'react-async-hook'
+import config from 'src/geth/networkConfig'
+
+export interface QueryParams {
+  address: string | null
+  localCurrencyCode: string
+  afterCursor?: string
+  dependencies: any[]
+  onSuccess: (result: any) => void
+  onError: (error: Error) => void
+  precondition?: () => boolean
+}
+
+export function useAsyncQueryTransactionsFeed(params: QueryParams) {
+  const {
+    address,
+    localCurrencyCode,
+    afterCursor,
+    dependencies,
+    onSuccess,
+    onError,
+    precondition,
```

Comment:
i was a bit confused about what a precondition is for, it's hard to give this a name 😅 maybe `shouldFetchMore`? or alternatively, since we're passing some specific props already, we could pass the `fetchMoreTransactions` and `pageInfo` directly?